scripts/cluster/create_multinode_cluster.go

```diff
@@ -141,24 +145,37 @@ func KubectlForNonRoot() error {
 func ExtractMasterNodeInfo() error {
 	// Extract master node information from logs
 	utils.WaitPrintf("Extracting master node information from logs")
+
+	// API Server address, port, token
 	shellOut, err := utils.ExecShellCmd("sed -n '/.*kubeadm join.*/p' < %s/masterNodeInfo | sed -n 's/.*join \\(.*\\):\\(\\S*\\) --token \\(\\S*\\).*/\\1 \\2 \\3/p'", configs.System.TmpDir)
-	if !utils.CheckErrorWithMsg(err, "Failed to extract master node information from logs!\n") {
+	if !utils.CheckErrorWithMsg(err, "Failed to extract API Server address, port, and token from logs!\n") {
 		return err
 	}
 	splittedOut := strings.Split(shellOut, " ")
 	configs.Kube.ApiserverAdvertiseAddress = splittedOut[0]
 	configs.Kube.ApiserverPort = splittedOut[1]
 	configs.Kube.ApiserverToken = splittedOut[2]
+
+	// API Server discovery token
 	shellOut, err = utils.ExecShellCmd("sed -n '/.*sha256:.*/p' < %s/masterNodeInfo | sed -n 's/.*\\(sha256:\\S*\\).*/\\1/p'", configs.System.TmpDir)
```

Comment:
These commands parsing `tmp/masterNodeInfo` return double lines of same info because in the output of `kubeadm init` there are two lines with `kubeadm join` (for control plane and workers). Because of that, masterKey.yaml is malformed.